src/codeu/chat/relay/Server.java

```diff
@@ -211,29 +211,23 @@ public boolean write(Uuid teamId,
 
     if (authenticate(teamId, teamSecret)) {
 
-      int remaining = Math.min(range, maxRead);
-
       LOG.info(
          "Request to read from server requested=%d allowed=%d",
           range,
           maxRead);
 
-      // Writing is a one way flag (once set it will not be unset) that switches
-      // between looking for the starting message and writing all messages to the
-      // output.
-      boolean writing = root == Uuid.NULL || root == null;
```

Comment:
Suggest:  () around comparison clauses (e.g., writing = (..) || (..) ) to improve readability.